src/wallet/wallet.cpp

```diff
@@ -3249,6 +3249,17 @@ std::unique_ptr<SigningProvider> CWallet::GetSolvingProvider(const CScript& scri
     return nullptr;
 }
 
+std::vector<WalletDescriptor> CWallet::GetWalletDescriptors(const CScript& script) const
+{
+    std::vector<WalletDescriptor> descs;
+    for (const auto spk_man: GetScriptPubKeyMans(script)) {
+        if (const auto desc_spk_man = dynamic_cast<DescriptorScriptPubKeyMan*>(spk_man)) {
+            descs.push_back(desc_spk_man->GetWalletDescriptor());
```

Comment:
Suggestion to avoid the error mentioned in https://github.com/bitcoin/bitcoin/pull/25504/files#r910264911.

```suggestion
            LOCK(desc_spk_man->cs_desc_man);
            descs.push_back(desc_spk_man->GetWalletDescriptor());
```